### PR TITLE
327-import-cut-off-1000-per-day: Add pageSize parameter in gaBatch request and override default of 1000

### PR DESCRIPTION
**Pull request by miiimooo**
### Description:

This is a quick fix patch to solve the issue of daily imports being limited to 1000 rows reported in #327 

The tacit default pageSize of 1000 is explicitly overwritten and set to 1 million.

In a next step this could be set in the config or support for a pager.

I've tested this with a local installation and GA data from a busy site with 10k URLs for one day and I get the correct number of rows in Matomo

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by AltamashShaikh** [comment]
@miiimooo Max 10K  would be returned no matter how high the no you set as per this [doc](https://developers.google.com/analytics/devguides/reporting/core/v3/reference#maxResults)
"The Analytics Core Reporting API returns a maximum of 10,000 rows per request, no matter how many you ask for."

**Comment by miiimooo** [comment]
@AltamashShaikh I think that must be a different endpoint. I've tested this and I get more than 10k rows results
image

Here is the endpoint the importer uses: https://developers.google.com/analytics/devguides/reporting/core/v4/rest/v4/reports/batchGet

> pageSize: Page size is for paging and specifies the maximum number of returned rows. Page size should be >= 0. A query returns the default of 1,000 rows. The Analytics Core Reporting API returns a maximum of 100,000 rows per request, no matter how many you ask for. It can also return fewer rows than requested, if there aren't as many dimension segments as you expect. For instance, there are fewer than 300 possible values for ga:country, so when segmenting only by country, you can't get more than 300 rows, even if you set pageSize to a higher value.

**Comment by AltamashShaikh** [comment]
@miiimooo Thanks I was checking the wrong one, `100,000` rows is really a good number and should work

**Comment by miiimooo** [comment]
@AltamashShaikh I mentioned the other endpoint in the issue #327 that was probably confusing

**Comment by mikkeschiren** [comment]
I have tested this patch, and it seems to work fine, solving the issue with limiting to 1000 rows. Got 13254 rows instead now.

**Comment by miiimooo** [comment]
I've changed the pageSize from 1m to 100k because that is the maximum according to the endpoint documentation. This is less confusing for future development

**Comment by AltamashShaikh** [comment]
@miiimooo Do you mind if I extend this PR or open a new PR where we can configure the row values via config and adjust the changes for testcase to work ?

**Comment by AltamashShaikh** [comment]
@miiimooo closing this in favour of #330

**Comment by AltamashShaikh** [comment]
@miiimooo / @mikkeschiren can you test the changes as per #330 ?

**Comment by miiimooo** [comment]
@AltamashShaikh I've hit our daily GA API limit when I tried to test :(

**Comment by AltamashShaikh** [comment]
@miiimooo Np, you can check as per your next working day and let us know how does it work 👍🏻

**Comment by AltamashShaikh** [comment]
@miiimooo Can you merge the latest `4.x-dev` branch changes in this branch and update the PR ?

**Comment by miiimooo** [comment]
I've merged in 4.x-dev - please review @AltamashShaikh

File: Google/GoogleAnalyticsQueryService.php
Comment (AltamashShaikh) [inline Google/GoogleAnalyticsQueryService.php]: @miiimooo can we remove this part ?

**Review by AltamashShaikh — COMMENTED** [review]
@miiimooo left 1 comment rest looks good to be merged

**Review by AltamashShaikh — APPROVED** [review]

File: Google/GoogleAnalyticsQueryService.php
Comment (miiimooo) [reply Google/GoogleAnalyticsQueryService.php]: I've removed it and pushed again

**Comment by AltamashShaikh** [comment]
@miiimooo Will just wait for testcases to pass once and merge it :+1: 
Thanks for the change :smile:

**Comment by miiimooo** [comment]
:pray:

**Pull request merged**
